backend/migrating/v2/migrate-to-v2.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# Usage: ./migrate-to-v2.sh
+
+set -o nounset # exit if a variable is not set
+set -o errexit # exit for any command failure"
+
+# text color escape codes (\033 == \e but OSX doesn't respect the \e)
+blue_text='\033[94m'
+green_text='\033[32m'
+red_text='\033[31m'
+default_text='\033[39m'
+yellow_text='\033[33m'
+
+# set -x/xtrace uses PS4 for more info
+PS4="$blue_text""${0}:${LINENO}: ""$default_text"
+
+# Function to check and create virtual environment
+setup_venv() {
+    if [[ ! -d ".venv" ]]; then
+        echo "Virtual environment not found. Creating one..."
+        pdm venv create -w virtualenv --with-pip || { echo "Failed to create virtual environment"; exit 1; }
+        pdm install || { echo "Failed to install dependencies"; exit 1; }
+    else
+        echo "Virtual environment found. Activating..."
+        source .venv/bin/activate || { echo "Failed to activate virtual environment"; exit 1; }
+    fi
+}
```

Comment:
@chandrasekharan-zipstack Suggest to remove this.
It needs to be done prior, and handled elsewhere.

Reply:
Okay, I'll exit with an error asking the user to setup venv if its not present